David/editing class and object layers

dwwop opened the pull request.

dwwop committed:
Merge branch 'networking' into david/editing-class-and-object-layers

dwwop committed:
Various UI fixes

- Remove duplicate FileBrowser
- Refactor Tool Manager
- PopUp setButtonsActive fix
- Class prefab change Name to unconstrained size

dwwop committed:
Adjust Namespaces to File Structure

dwwop committed:
Deep edit, user input validation and bug fixes

Deep edit of Class Names
Zmena generovania idciek
Osetrenie uzivatelskeho vstupu
Oprava error labels

dwwop committed:
Merge branch 'networking' into david/editing-class-and-object-layers

Martin554 commented on UnityProjectDP/Assets/Scripts/UMSAGL/Scripts/LineControl.cs:
```diff
@@ -1,19 +0,0 @@
-﻿using AnimArch.Visualization.UI;
```
toto sa preco zmazalo?

dwwop replied:
Pretoze to tak nevie a nebude fungovat, vzhladom na to ako funguje LineRenderer. Myslim, ze to len niekto skusal a pushol, ale je to nefunkcne

Martin554 commented on UnityProjectDP/Assets/Scripts/Visualization/Animation/Anim.cs:
```diff
@@ -1,12 +1,18 @@
+//Filip
 
-namespace AnimArch.Visualization.Animating
+//Filip
```
toto mozes zmazat

Martin554 commented on UnityProjectDP/Assets/Scripts/Visualization/Animation/AnimationData.cs:
```diff
@@ -27,12 +27,16 @@ public List<Anim> getAnimList()
         {
             animations.Add(anim);
+            GameObject.Find("MainPanel").transform.Find("Edit").GetComponentInChildren<Button>().interactable = true;
+            GameObject.Find("MainPanel").transform.Find("Play").GetComponentInChildren<Button>().interactable = true;
```
Toto mozno vytiahnut niekam do UI kazde do svojej metody s atributom pre bool interactable. nieco v zmysle 
```suggestion
void EditButtonInteractable(bool interactable)
{
GameObject.Find("MainPanel").transform.Find("Edit").GetComponentInChildren<Button>().interactable = interactable;
 }
```
pripadne
```suggestion
void SetButtonInteractable(bool interactable, string buttonName)
{
GameObject.Find("MainPanel").transform.Find(buttonName).GetComponentInChildren<Button>().interactable = interactable;
 }
```

Martin554 commented on UnityProjectDP/Assets/Scripts/Visualization/ClassDiagram/ClassComponents/Method.cs:
```diff
@@ -28,5 +26,21 @@ public Method(string name, string id, string returnValue)
+
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(Id, Name);
```
tu sa netreba pytat, ako pri ostatnych na null?

dwwop replied:
Hashcode.combine by mal vediet pracovat aj s Null. Ale tbh nechal som si tie metody automaticky vygenerovat IDEckom

dwwop committed:
Bugfixes and improvements

Error popUp when creating relation
Improved error labels with more errors
Tool Manager Reset Button
Enter on Input field to add component from popup
Ctrl + mousewheel to zoom

Martin554 commented on UnityProjectDP/Assets/Scripts/Visualization/Animation/Animation.cs:
```diff
@@ -608,18 +608,25 @@ private void HighlightObjectRelation(long callerInstanceId, long calledInstanceI
+            if (objectRelation == null)
+                return;
+            
+            var objectRelationGO = objectRelation.GameObject;
```
```suggestion
            var objectRelationGo = objectRelation.GameObject;
```

Martin554 commented on UnityProjectDP/Assets/Scripts/Visualization/ClassDiagram/Diagrams/ClassDiagram.cs:
```diff
@@ -111,6 +110,14 @@ public string FindOwnerOfRelation(string relationName)
                 .FirstOrCustomDefault(
                     relationInDiagram => relationInDiagram.ParsedRelation.FromClass, "");
         }
+        public RelationInDiagram FindRelationWithType(string fromClass, string toClass, string type)
```
asi staci takto
```suggestion
        public RelationInDiagram FindRelation(string fromClass, string toClass, string type)
```

Martin554 commented on UnityProjectDP/Assets/Scripts/Visualization/UI/PopUps/AbstractClassPopUp.cs:
```diff
@@ -1,21 +1,46 @@
 {
     public abstract class AbstractClassPopUp : AbstractPopUp
     {
+        protected const string ErrorEmptyName = "Name can't be empty!";
```
```suggestion
        protected const string ErrorEmptyName = "Name can not be empty!";
```

Martin554 commented on UnityProjectDP/Assets/Scripts/Visualization/UI/PopUps/AbstractTypePopUp.cs:
```diff
@@ -1,26 +1,32 @@
     public abstract class AbstractTypePopUp : AbstractClassPopUp
     {
-        private const string CUSTOM = "custom";
+        private const string ErrorTypeEmpty = "Type can't be empty";
```
```suggestion
        private const string ErrorTypeEmpty = "Type can not be empty";
```

Martin554 commented on UnityProjectDP/Assets/Scripts/Visualization/UI/PopUps/AbstractTypePopUp.cs:
```diff
@@ -1,26 +1,32 @@
         private readonly HashSet<TMP_Dropdown.OptionData> _variableData = new();
 
-        private void Awake()
+        protected new void Awake()
```
preco new?

dwwop replied:
The new modifier creates a new member with the same name and causes the original member to become hidden. 

Basically to funguje jak override pre metody co nie su abstract/virtual. Thats my understanding + IDE frflalo

Martin554 commented on UnityProjectDP/Assets/Scripts/Visualization/UI/PopUps/AttributePopUp.cs:
```diff
@@ -45,30 +47,38 @@ public override void Confirmation()
                     return;
                 }
 
+                newAttribute.Id = DateTimeOffset.Now.ToUnixTimeMilliseconds().ToString();
```
nebolo by lepsie pouzit pre vestky ID-cka UUID? https://learn.microsoft.com/en-us/dotnet/api/system.guid.newguid?redirectedfrom=MSDN&view=net-7.0#System_Guid_NewGuid
to len ako navrh

dwwop replied:
Vedel som ze take nieco musi existovat, ale nechcelo sa mi googlit. Zakomponujem to

Martin554 commented on UnityProjectDP/Assets/Scripts/Visualization/UI/ToolManager.cs:
```diff
@@ -66,10 +63,56 @@ public void SelectColor(string colorID)
+            clickedButton.interactable = false;
+        }
+
+        private void SelectTool(Tool toolName)
```
```suggestion
        private void SelectTool(Tool tool)
```
nie je to meno, ale tool (enum)

dwwop committed:
Changes based on comments from pull req
